carma-messenger-core/cpp_message/src/MobilityRequest_Message.cpp

```diff
@@ -0,0 +1,538 @@
+        }
+        message->value.choice.TestMessage00.header.timestamp.buf=string_content_timestamp;
+        message->value.choice.TestMessage00.header.timestamp.size=string_size;
+        //strategy
+        string_size=plainMessage.strategy.size();
+        uint8_t string_content_strategy[string_size];
+        for(size_t i=0;i<string_size;i++)
+        {
+            string_content_strategy[i]=plainMessage.strategy[i];
+        }
+        message->value.choice.TestMessage00.body.strategy.buf=string_content_strategy;
+        message->value.choice.TestMessage00.body.strategy.size=string_size;
+
+        //plantype
+        MobilityPlanType_t type;
+            enum plan {UNKNOWN=0,CHANGE_LANE_LEFT=1,CHANGE_LANE_RIGHT=2,JOIN_PLATOON_AT_REAR=3,PLATOON_FOLLOWER_JOIN=4};
+            //plan plan_type;
+            uint8_t plan_type=plainMessage.plan_type.type;
+            switch(plan_type){
+                case 0:  
+                    type=UNKNOWN;
+                    break;
+                case 1:
+                    type=CHANGE_LANE_LEFT;
+                    break;
+                case 2:
+                    type=CHANGE_LANE_RIGHT;
+                    break;
+                case 3:
+                    type=JOIN_PLATOON_AT_REAR;
+                    break;
+                case 4:
+                    type=PLATOON_FOLLOWER_JOIN;
+                    break;
+                default:type=UNKNOWN;
+           }
+            
+        message->value.choice.TestMessage00.body.planType=type;
+
+        //urgency
+        message->value.choice.TestMessage00.body.urgency=plainMessage.urgency;
+
+        //location
+        long location_val;
+        location_val=plainMessage.location.ecef_x;
+        if(location_val> LOCATION_MAX || location_val<LOCATION_MIN){
+            ROS_WARN_STREAM("Location ecefX is out of range");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        message->value.choice.TestMessage00.body.location.ecefX=location_val;
+
+        location_val=plainMessage.location.ecef_y;
+        if(location_val> LOCATION_MAX || location_val<LOCATION_MIN){
+            ROS_WARN_STREAM("Location ecefY is out of range");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        message->value.choice.TestMessage00.body.location.ecefY=location_val;
+
+        location_val=plainMessage.location.ecef_z;
+        if(location_val> LOCATION_MAX_Z || location_val<LOCATION_MIN_Z){
+            ROS_WARN_STREAM("Location ecefX is out of range");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        message->value.choice.TestMessage00.body.location.ecefZ=location_val;
+
+        uint64_t location_time=plainMessage.location.timestamp;
+        std::string location_timestamp=std::to_string(location_time);
+        size_t location_timestamp_string_size=location_timestamp.size();
+        uint8_t string_content_location_timestamp[location_timestamp_string_size];
```

Comment:
There might be a way to do this with std::strcpy or using the std::string::c_str() method to get the raw byte array and then using std::copy, but this works for now unless you feel like investigating that.